core/src/main/java/org/bitcoinj/core/BlockLocator.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2018.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.bitcoinj.core;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * Represents Block Locator in GetBlocks and GetHeaders messages
+ **/
+public final class BlockLocator {
+    private final ImmutableList<Sha256Hash> hashes;
+
+    public BlockLocator() {
+        hashes = ImmutableList.of();
+    }
+
+    /**
+     * creates a Block locator with defined list of hashes
+     * @param hashes
+     */
+    public BlockLocator(ImmutableList<Sha256Hash> hashes) {
+        this.hashes = hashes;
+    }
+
+    /**
+     * Add a {@link Sha256Hash} to newly created block locator
+     *
+     * @param hash
+     * @return
+     */
+    public BlockLocator add(Sha256Hash hash) {
+        return new BlockLocator(new ImmutableList.Builder<Sha256Hash>().addAll(this.hashes).add(hash).build());
+    }
+
+    /**
+     * returns size of block locator list
+     *
+     * @return
+     */
+    public int size() {
+        return hashes.size();
+    }
+
+    /**
+     * returns {@link List<Sha256Hash>} of Block locator hashes
+     *
+     * @return
+     */
+    public List<Sha256Hash> getHashes() {
+        return hashes;
+    }
+
+    /**
+     * compare list of hashes of two block locators
```

Comment:
But shouldn't this then be named `equals()` (effectively overloading the other equals)

`containsAll()` implies that order doesn't matter, doesn't it?